src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1107,13 +1098,23 @@ private void startConnect() throws IOException {
             String hostPort = addr.getHostString() + ":" + addr.getPort();
             MDC.put("myid", hostPort);
             setName(getName().replaceAll("\\(.*\\)", "(" + hostPort + ")"));
-            if (ZooKeeperSaslClient.isEnabled()) {
+            ZKClientConfig conf = zooKeeper.getClientConfig();
```

Comment:
I'm wondering if is better to have an object attribute `ZKClientConfig conf` that is assigned upon construction rather than having local variables or keep calling `zookeeper.getClientConfig()`.